clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -142,6 +143,22 @@ struct BinOpInfo {
       return UnOp->getSubExpr()->getType()->isFixedPointType();
     return false;
   }
+
+  /// Check if at least one operand is a decimal float type. In such cases, this
+  /// operation did not follow usual arithmetic conversion and both operands
+  /// might not be of the same type.
```

Comment:
I wasn't sure if we should have an invariant that both types be DFP types or both not be, so I checked the Intel DFP library and I don't see any functions exposed to perform operations between binary and DFP types. That suggests to me that we should have such an invariant.